lambdas/services/dynamo_query_service.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+import boto3
+from boto3.dynamodb.conditions import Key
+from botocore.exceptions import ClientError
+
+from enums.metadata_field_names import DynamoField
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+ALL_FIELDS = "ID, ContentType, Created, DocumentUploaded, FileName, Indexed, Location, NhsNumber, Type, VirusScanResult"
+
+
+class DynamoQueryService:
+    def __init__(self, table_name):
+        self.TABLE_NAME = table_name
+
+    def __call__(self, nhs_number: str, requested_fields: list = None):
+        try:
+            dynamodb = boto3.resource('dynamodb')
+            table = dynamodb.Table(self.TABLE_NAME)
+
+            if requested_fields is None:
+                requested_fields = DynamoField.list()
+
+            projection_expression, expression_attribute_names = self.create_expressions(requested_fields)
+
+            return table.query(
+                IndexName='NhsNumberIndex',
```

Comment:
Maybe pass the IndexName into the service so we can use the service for other dynamo tables? (same with the key and ALL_FIELDS

Reply:
Spoke to Noga and sorted, done.